src/get-started/huff-by-example/README.md

```diff
@@ -491,4 +491,15 @@ somewhere within the contract.
 #define table CODE_TABLE {
     0x604260005260206000F3
 }
+
+#define macro MAIN() = takes (0) returns (0) {
+  __tablesize(CODE_TABLE)  // [size]
+  // copy table to memory
+  dup1                     // [size, size]
+  __tablestart(CODE_TABLE) // [start, size, size]
+  returndatasize           // [0, start, size, size]
+  mstore                   // [size]
+  returndatasize           // [0, size]
+  return 
+}
```

Comment:
Don't think this works as intended- maybe something like this?

```js
#define macro MAIN() = {
    __tablesize(CODE_TABLE)  // [table_size]
    __tablestart(CODE_TABLE) // [table_start, table_size]
    returndatasize           // [0x00, table_start, table_size]
    codecopy                 // []
    msize returndatasize     // [0x00, 0x20]
    return                   // []
}
```

Also wondering if we should use tricks like `returndatasize` here- idea is to have easily readable (er, as easily readible as we can manage) code here.

Reply:
Yeah my bad, replace `mstore` with `codecopy` in my example and it'll work (compiled and tested this time), your example would pad the return with 0-bytes but it works too I guess.

When it comes to tricks like `returndatasize` I feel like they should at least be in the docs somewhere but I agree that this might not be the best place. My thought was that if someone sees that and actually thinks to question it they'll learn as they research / ask about it.